### PR TITLE
Add solid body rotation test

### PR DESCRIPTION
**Pull request by jiahe23**
Need:
- [ ] discrete hydrostatic balance

### CODE REVIEW

File: examples/3dsphere/solid_body_rotation_3d.jl [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+# initial conditions for density and total energy
+function init_sbr_thermo(z)
+
+    p = p_0 * exp(-z * grav / R_d / T_0)
+    ρ = 1 / R_d / T_0 * p
+
+    T = p / ρ / R_d
+
+    e = cv_d * (T - T_tri) + grav * z
```
Comment (szy21) [inline examples/3dsphere/solid_body_rotation_3d.jl:86]:
```suggestion
    e = cv_d * (T - T_tri) + Φ(z)
```

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -448,6 +448,17 @@ steps:
...
+      config: cpu
+      queue: central
+      slurm_ntasks: 1
+  
+  - label: ":computer: solid body rotation"
+    key: "cpu_solid_body_rotation"
+    command:
+      - "julia --color=yes --project=examples/3dsphere examples/3dsphere/solid_body_rotation.jl"
+    artifact_paths:
+      - "examples/3dsphere/output/deformation_flow/*"
```
Comment (szy21) [inline .buildkite/pipeline.yml:461]:
```suggestion
      - "examples/3dsphere/output/solid_body_rotation/*"
```

Comment (jiahe23) [reply .buildkite/pipeline.yml:461]: lol

File: examples/3dsphere/solid_body_rotation_3d.jl [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+function pressure(ρ, e, normuvw, z)
+    I = e - Φ(z) - normuvw^2 / 2
+    T = I / cv_d + T_tri
+    return ρ * R_d * T
+end
+
+# initial conditions for density and total energy
+function init_sbr_thermo(z)
+
+    p = p_0 * exp(-z * grav / R_d / T_0)
```
Comment (szy21) [inline examples/3dsphere/solid_body_rotation_3d.jl:81]:
```suggestion
    p = p_0 * exp(-z / H)
```
Or you can also remove const H?

**Review by szy21 — APPROVED** [review]

**Comment by jiahe23** [comment]
bors try

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -448,6 +448,17 @@ steps:
...
       - "examples/3dsphere/output/baroclinic_wave/*"
+    agents:
+      config: cpu
+      queue: central
+      slurm_ntasks: 1
+  
+  - label: ":computer: solid body rotation"
+    key: "cpu_solid_body_rotation"
+    command:
+      - "julia --color=yes --project=examples/3dsphere examples/3dsphere/solid_body_rotation.jl"
```
Comment (szy21) [inline .buildkite/pipeline.yml:459]:
```suggestion
      - "julia --color=yes --project=examples/3dsphere examples/3dsphere/solid_body_rotation_3d.jl"
```

**Comment by jiahe23** [comment]
bors try

**Pull request merged**